scripts/check_compliance.py

```diff
@@ -588,6 +588,118 @@ def run(self):
+               "E1004,"  # Missing argument to super()
+               #"E1101,"  # Non-existent object member
+               "E1102,"  # Not callable
+               #"E1103,"  # Non-existent object member, more complex case (?)
+               "E1111,"  # Assigning to function call that doesn't return
+               "E1122,"  # Missing argument in function call
+               #"E1123,"  # Bad keyword argument in function call
+               "E1124,"  # Argument passed both positionally and as keyword
+               "E1200,"  # Unsupported logging format character
+               "E1201,"  # Logging format string ends in middle of conversion specifier
+               "E1205,"  # Too many arguments for logging format string
+               "E1206,"  # Not enough arguments for logging format string
+               "E1300,"  # Unsupported format character
+               "E1301,"  # Format string ends in middle of conversion specifier
+               "E1302,"  # Mixed named and unnamed conversion specifiers
+               "E1303,"  # Expected mapping for format string, not %s
+               "E1304,"  # Missing key in format string dictionary
+               "E1305,"  # Too many arguments for format string
+               "E1306,"  # Not enough arguments for format string
+               "E1310,"  # Suspicious argument in call
+               "R1711,"  # Redundant 'return'
+               "W0105,"  # String statement has no effect
+               "W0611,"  # Unused import
+               "W0612,"  # Unused variable
+               "W0301,"  # Unnecessary semicolon
+               "W0311,"  # Bad indentation
+               "W1401,"  # Backslash in non-r string
+              )
+
+        py_files = self.py_files()
+        try:
+            # Work around a crash in pylint 2.2.2:
+            # https://github.com/PyCQA/pylint/issues/2906
+            py_files.remove("boards/xtensa/intel_s1000_crb/support/create_board_img.py")
+        except ValueError:
+            self.error("Workaround in check_compliance.py no longer needed? "
+                       "Please check (grep for this message).")
+
+        process = subprocess.Popen(cmd.split() + py_files,
+                                   stdout=subprocess.PIPE,
+                                   stderr=subprocess.PIPE,
+                                   cwd=ZEPHYR_BASE)
+
+        stdout, stderr = process.communicate()
+        if process.returncode or stderr:
+            # Issues found, or a problem with pylint itself
+            self.add_failure(stdout.decode("utf-8") + stderr.decode("utf-8"))
+
+
+    def py_files(self):
+        """
+        Returns a list with the paths of all .py files, relative to the Zephyr
+        root.
+
+        Uses 'git ls-files', which is fast and avoids picking up any files that
+        are not checked in.
+        """
+        cmd = "git ls-files -z *.py"  # -z = null-separate entries
+
+        process = subprocess.Popen(cmd.split(),
```

Comment:
Anything wrong with subprocess.check_output()?

Reply:
Also please test and make sure a missing pylint doesn't stop other, unrelated checks.

Reply:
Tried a rewrite now.

One thing that gets less clean compared to `Popen()` is failing if there's anything on `stderr`, even if the return status is 0, without redirecting `stderr`.

No value in using `check_output()` for its own sake, IMO, if `Popen()` gives you a little bit if added flexibility that you make use of. The `Popen()` interface isn't all that bad either.

Reply:
Now checks for troubles starting `pylint`.

Realized even `subprocess.check_output()` raises `OSError` if the executable is missing btw, so there's probably other places that ought to catch it.